From f0ff41893892ab560e8fa4f90b2dd35bee7d2595 Mon Sep 17 00:00:00 2001
From: magic0whi <lollipop.studio.cn@gmail.com>
Date: Fri, 8 Mar 2019 09:47:38 +0000
Subject: [PATCH] arm64: dts: rockchip: add more cpu operating points for
 roc-rk3328-cc

This allows for greater max frequency on rk3328 boards,
increasing performance.

It has been included in Armbian (a linux distibution for ARM boards)
for a while now without any reported issues

https://github.com/armbian/build/blob/master/patch/kernel/rockchip64-default/enable-1392mhz-opp.patch
https://github.com/armbian/build/blob/master/patch/kernel/rockchip64-default/enable-1512mhz-opp.patch
https://patchwork.kernel.org/patch/10836075/

Signed-off-by: magic0whi <lollipop.studio.cn@gmail.com>
---
 arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts b/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts
index 7f14d5039..5af00e47a 100644
--- a/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3328-roc-cc.dts
@@ -120,6 +120,19 @@
 	cpu-supply = <&vdd_arm>;
 };
 
+&cpu0_opp_table {
+		opp-1392000000 {
+			opp-hz = /bits/ 64 <1392000000>;
+			opp-microvolt = <1350000>;
+			clock-latency-ns = <40000>;
+		};
+		opp-1512000000 {
+			opp-hz = /bits/ 64 <1512000000>;
+			opp-microvolt = <1450000>;
+			clock-latency-ns = <40000>;
+		};
+};
+
 &emmc {
 	bus-width = <8>;
 	cap-mmc-highspeed;
-- 
2.21.0

